src/backend/distributed/commands/schema.c

```diff
@@ -153,6 +157,57 @@ PreprocessGrantOnSchemaStmt(Node *node, const char *queryString)
 }
 
 
+/*
+ * PreprocessAlterSchemaRenameStmt is called when the user is renaming a schema.
+ * The invocation happens before the statement is applied locally.
+ *
+ * As the schema already exists we have access to the ObjectAddress for the schema, this
+ * is used to check if the schmea is distributed. If the schema is distributed the rename
+ * is executed on all the workers to keep the schemas in sync across the cluster.
+ */
+List *
+PreprocessAlterSchemaRenameStmt(Node *node, const char *queryString)
+{
+	ObjectAddress schemaAddress = GetObjectAddressFromParseTree(node, false);
+	if (!ShouldPropagateObject(&schemaAddress))
+	{
+		return NIL;
+	}
+
+	/* deparse sql*/
+	const char *renameStmtSql = DeparseTreeNode(node);
```

Comment:
I'd like a call to `QualifyTreeNode` above this to make the function more similar to the other `Prepare...Stmt` functions. Even though we don't have a qualify function on this type of tree - which is correct -. It will be effectively a no-op.

Reasons is that this prepare function becomes so generic we can at some point fold them into 1 so all statements behave the same. Already having this call here makes it easier to merge it with others.